src/lib/clk.c

```diff
@@ -152,6 +152,11 @@ uint64_t clock_ms_to_ticks(int clock, uint64_t ms)
 	return clk_pdata->clk[clock].ticks_per_msec * ms;
 }
 
+uint64_t clock_ns_to_ticks(int clock, uint64_t ns)
+{
+	return clk_pdata->clk[clock].ticks_per_msec / 1000 * ns / 1000;
```

Comment:
@bardliao 
Maybe this way?
```suggestion
	return clk_pdata->clk[clock].ticks_per_msec * ns / 1000000;
```

Reply:
@xiulipan It is a trade-off between overflow and accuracy.  It may overflow when ``clk_pdata->clk[clock].ticks_per_msec * ns``

Reply:
Then maybe add a comment here to explain the overflow risk.

Reply:
@bardliao would it be easier to use the clock here in this calculation since it will be in MHz (multiplied by 1000 it will be same timebase as host ns)  ?

Reply:
@lgirdwood Sorry I didn't get you.  I think the ``clock`` here is an index, right? Did you mean something like [double clock;](https://github.com/thesofproject/sof/blob/master/tools/logger/convert.h#L28) in logger? I didn't find a similar variable here.